analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -344,13 +344,15 @@ private fun resolveConfiguration(
     val packageIds = analyzerResult.packages.mapTo(mutableSetOf()) { it.id }
     val packageCurations = mutableListOf<PackageCuration>()
 
-    val duration = measureTime {
-        curationProviders.forEach { (_, curationProvider) ->
-            packageCurations += curationProvider.getCurationsFor(packageIds).values.flatten()
+    curationProviders.forEach { (id, curationProvider) ->
+        val (curations, duration) = measureTimedValue {
+            curationProvider.getCurationsFor(packageIds).values.flatten()
         }
-    }
 
-    Analyzer.logger().debug { "Getting package curations took $duration." }
+        packageCurations += curations
+
+        Analyzer.logger().info { "Getting ${curations.size} package curations from provider '$id' took $duration." }
+    }
 
-    return ResolvedConfiguration(packageCurations)
+    return ResolvedConfiguration(packageCurations.distinct())
```

Comment:
Hmm... besides this being a functional change compared to before, I wondering if we can safely do that. Assume curations A and B, where B undoes A. Then having A, B, A *without* `distinct()` would result in A, whereas *with* `distinct()` it would result A, B (which would be a no-op).

Reply:
Hm, I've read your message again. So, if we agree on "no order defined within single provider" then we should have
one distinct call per provider.

Reply:
> So, if we agree on "no order defined within single provider" then we should have one distinct call per provider. Agree?

I'm not sure yet. While thinking about it, I'd to understand a bit more about the actual problem you're trying to solve: Did you really run into the case of unwanted duplicate curations, or is this just a theoretical thing for now?

Reply:
BTW, I just came across 8aeb20566e1404979d7a3667d3de2e4f48e319d3, so I guess it's also ok to do that for all providers (and in exchange remove that code from `ClearlyDefinedPackageCurationProvider`).

Reply:
BTW, I just came across 8aeb20566e1404979d7a3667d3de2e4f48e319d3, so I guess it's also ok to do that for all providers (and in exchange remove that code from `ClearlyDefinedPackageCurationProvider`).

Reply:
> I'm not sure yet. While thinking about it, I'd to understand a bit more about the actual problem you're trying to solve: Did you really run into the case of unwanted duplicate curations, or is this just a theoretical thing for now?

When implementing the separation of curations and packages in analyzer result, one of the reasons / benefits was to me always that it would drop redundant curations, because a curation may omit the `name` part of the `identifier`. I just forgot about adding this distinct call. So, given that a curation can match multiple package I believe it's not theoretical. Practical use case I remember: Add metadata to all curations under the android standard packages namespace.

Reply:
> BTW, I just came across [8aeb205](https://github.com/oss-review-toolkit/ort/commit/8aeb20566e1404979d7a3667d3de2e4f48e319d3), so I guess it's also ok to do that for all providers (and in exchange remove that code from `ClearlyDefinedPackageCurationProvider`).

Ok, so I would keep the code as-is right now (after my recent force push), and add a commit on top which drops the logic from clearly defined. Would that be the correct understanding of your proposal?

Reply:
Hm, from the perspective of the `ClearlyDefinedProvider` (without its usages from within ORT code) I believe it makes sense to keep the duplicate filtering there, in particular for programmatic use of the curation provider.
If you are fine let's keep it there just for now:

Note: I'll propose (PR) a simplification of the provider API (on top of your API change) which might bring a bit more clarity if it gets accepted.